src/librustc_infer/infer/resolve.rs

```diff
@@ -227,7 +227,7 @@ impl<'a, 'tcx> TypeFolder<'tcx> for FullTypeResolver<'a, 'tcx> {
             match c.val {
                 ty::ConstKind::Infer(InferConst::Var(vid)) => {
                     self.err = Some(FixupError::UnresolvedConst(vid));
-                    return self.tcx().consts.err;
+                    return self.tcx().mk_const(ty::Const { val: ty::ConstKind::Error, ty: c.ty });
```

Comment:
Here and in the other the places... (method doesn't exist ofc, so will need to be made)
```suggestion
                    return self.tcx().mk_const_err(c.ty);
```

Reply:
> Wouldn't it be better to silence all further errors by using `tcx.types.err` for such constants?

No, that loses miri the ability to do sanity checks before dismissing the constant as an error.
IME, it can result in ICEs, although I don't have a good example on hand.

Reply:
Okay, looking at the commit list again, I think I tried using `tcx.consts.err` before and that ICEd, so I decided to do all of this work *so that* the type is known.